api/system/services/System.js

```diff
@@ -1,8 +1,106 @@
 'use strict';
+const fetch = require('node-fetch');
+
 
 /**
  * Read the documentation (https://strapi.io/documentation/3.0.0-beta.x/guides/services.html#core-services)
  * to customize this service
  */
 
-module.exports = {};
+module.exports = {
+  /**
+   * Promise to add record
+   *
+   * @return {Promise}
+   */
+
+  create: async (values) => {
+    let edsmData = await strapi.services.system.edsmUpdate(values);
+
+    return strapi.query('System').create(edsmData);
+  },
+
+  /**
+   * Promise to edit record
+   *
+   * @return {Promise}
+   */
+
+  update: async (params, values) => {
+
+    if (!values.edsmCoordLocked){
+      let oldData = await strapi.services.system.findOne({id: params.id});
```

Comment:
Same here, try/cath, together with `oldData = oldData.toJSON();` as `toJSON()` can throw some parsing errors.

Reply:
the `toJSON` is a temporary thing from bookshelf to convert the strapi response back to json from it's prototype object